libs/libloggers/loggers/OwnPatternFormatter.cpp

```diff
@@ -4,15 +4,72 @@
 #include <optional>
 #include <IO/WriteBufferFromString.h>
 #include <IO/WriteHelpers.h>
+#include <Common/HashTable/Hash.h>
 #include <Interpreters/InternalTextLogsQueue.h>
 #include <sys/time.h>
 #include <Common/CurrentThread.h>
 #include <common/getThreadId.h>
 #include "Loggers.h"
 
 
-OwnPatternFormatter::OwnPatternFormatter(const Loggers * loggers_, OwnPatternFormatter::Options options_)
-    : Poco::PatternFormatter(""), loggers(loggers_), options(options_)
+static const char * setColor(UInt64 num)
+{
+    /// ANSI escape sequences to set foreground font color in terminal.
+
+    static constexpr auto num_colors = 14;
+    static const char * colors[num_colors] =
+    {
+        /// Black on black is meaningless
+        "\033[0;31m",
+        "\033[0;32m",
+        "\033[0;33m",
+        /// Low intense blue on black is too dark.
+        "\033[0;35m",
+        "\033[0;36m",
+        "\033[0;37m",
+        "\033[1;30m",
+        "\033[1;31m",
+        "\033[1;32m",
+        "\033[1;33m",
+        "\033[1;34m",
+        "\033[1;35m",
+        "\033[1;36m",
+        "\033[1m",  /// Not as white but just as high intense - for readability on white background.
+    };
+
+    return colors[num % num_colors];
+}
+
+static const char * setColorForLogPriority(int priority)
+{
+    if (priority < 1 || priority > 8)
+        return "";
+
+    static const char * colors[] =
+    {
+        "",
+        "\033[1;41m",   /// Fatal
+        "\033[0;41m",   /// Critical
+        "\033[1;31m",   /// Error
+        "\033[0;31m",   /// Warning
+        "\033[0;33m",   /// Notice
+        "\033[1m",      /// Information
+        "",             /// Debug
+        "\033[1;30m",   /// Trace
```

Comment:
Is black for the trace level is the intention? (With my urxvt+solarized it is empty)

Reply:
It's dark gray, looks great on black background... but for solarized - what we can do?
Maybe we can use something from some neutral subset of 256-color pallete?

Also I have thoughts that style change in terminal also remaps some of the 16 colors to avoid this effect...

Reply:
Or: force black background for all colors.

Reply:
>It's dark gray, looks great on black background... but for solarized - what we can do?

non-dark gray (`\033[0;30m`) is ok

>Or: force black background for all colors.

But I guess this will be better